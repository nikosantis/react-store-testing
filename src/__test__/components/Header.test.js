import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Header from '../../components/Header';

describe('<Header />', () => {
  test('Render of Header component', () => {
    const header = shallow(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.length).toEqual(1);
  });
  test('Render of title', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.find('.Header-title').text()).toEqual('React Store');
  });
});

describe('Header Snapshot', () => {
  test('Check the UI of the Header component', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
