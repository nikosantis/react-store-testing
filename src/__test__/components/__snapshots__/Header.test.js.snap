// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Check the UI of the Header component 1`] = `
<div
  className="Header"
>
  <h1
    className="Header-title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      React Store
    </a>
  </h1>
  <div
    className="Header-checkout"
  >
    <a
      href="/checkout"
      onClick={[Function]}
    >
      <i
        className="fas fa-shopping-basket"
      />
    </a>
  </div>
</div>
`;
